test/shared/src/main/scala/zio/test/mock/Mockable.scala

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright 2017-2019 John A. De Goes and the ZIO Contributors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package zio.test.mock
+
+trait Mockable[Interface] {
+
+  def make(mock: Mock): Interface
+}
+
+object Mockable {
+
+  def apply[Interface](implicit mockable: Mockable[Interface]): Mockable[Interface] =
```

Comment:
I think quite possibly we should not use the type class pattern for `Mockable`. That is, no implicits, make everything explicit. It doesn't really help us that much since you only need to provide a `Mockable` when a spec is run, not when it's constructed, so it's almost no savings.

Also the way the code is organized, the instances would not be found, anyway. So the user would need to import them implicitly, which is an anti-pattern.

Reply:
Hmm I thought companion objects of type arguments (here `Interface`) should also be searched.
Will check if I can figure something out to fix this. If its not possible then I agree - no point in it being implicit.

Reply:
@jdegoes I was right, the problem was the `managedEnv[I]` contained an abstract type `I` that compiler had no idea how to fill (therefore it had no idea where to look for implicits). Fixed the issue in latest commit :grin: